src/core/MOM.F90

```diff
@@ -1436,8 +1436,11 @@ subroutine step_MOM_tracer_dyn(CS, G, GV, US, h, Time_local)
                       CS%tracer_diff_CSp, CS%tracer_Reg, CS%tv)
   if (CS%debug) call MOM_tracer_chksum("Post-diffuse ", CS%tracer_Reg, G)
   if (showCallTree) call callTree_waypoint("finished tracer advection/diffusion (step_MOM)")
-  call update_segment_tracer_reservoirs(G, GV, CS%uhtr, CS%vhtr, h, CS%OBC, &
-                     CS%t_dyn_rel_adv, CS%tracer_Reg)
+  if (associated(CS%OBC) .and. allocated(CS%uhtr) .and. allocated(CS%vhtr)) then
```

Comment:
This will likely trigger the same problem as PR 520. As Marshall recommended it can be changed to:
 `if (associated(CS%OBC)) then ...`